oaknational/Oak-Web-Application
# build: prettier, eslintconfig, tsconfig

## Description

- adds prettier
- adds next eslint plugin, following instruction here https://nextjs.org/docs/basic-features/eslint#recommended-plugin-ruleset
- adds import eslint plugin, enabling following rules:
  -  `"@typescript-eslint/no-unused-vars": "error"`
  -  `"@typescript-eslint/no-explicit-any": "error"`
- adds typescript eslint plugin, enabling the following rules:
  - `"import/first": "error",`
  - `"import/no-unresolved": "error",`
  - `"import/order": "error"`
- enables following rules in tsconfig:
  - `"noUncheckedIndexedAccess": true,`
  - `"noImplicitAny": true`
- adds `format` script which runs `prettier --write`
- runs `format` script, which is why there are the semi-colons and double-quotes in the diff

## Issues

Fixes #2 

## How to test

- E.g. inside a component eg `_app.tsx`, try to import a named import (not sure why it doesn't work for non named imports such as the css file) locally before `import type { AppProps } from "next/app";`
- Eslint should complain in your text editor
- Run `npm run lint` 
- It should fail citing rule `import/order`.
